src/js/components/CosmosErrorMessage.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+const REPOSITORY_ERRORS = [
+  'EmptyPackageImport',
+  'IndexNotFound',
+  'PackageFileMissing',
+  'PackageFileNotJson',
+  'RepositoryNotPresent',
+  'RepositoryUriConnection',
+  'RepositoryUriSyntax'
+];
+
+class CosmosErrorMessage extends React.Component {
+  getHeader() {
+    let {header, headerClass} = this.props;
+    if (!header) {
+      return null;
+    }
+
+    return (
+      <span className={headerClass}>
+        {header}
+      </span>
+    );
+  }
+
+  getMessage() {
+    let {type, message} = this.props.error;
+
+    // Append reference to repository page, since repository related errors
+    // can occur at any request to Cosmos
+    if (REPOSITORY_ERRORS.includes(type)) {
+      message = this.appendRepositoryLink(message);
+    }
+
+    return message;
+  }
+
+  appendRepositoryLink(message) {
+    return (
+      <span>
+        {`${message} You can go to the `}
+        <a href="/#/system/overview/repositories/">Repositories Settings</a>
```

Comment:
It's not since this will be used in the modal too. So I'd rather have this work everywhere we use it, rather than adding context to modal somehow.